plugins/OnlineSales.Plugin.SendGrid/Tasks/SyncSuppressionsTask.cs

```diff
@@ -46,57 +48,56 @@ public SyncSuppressionsTask(TaskStatusService taskStatusService, PgDbContext dbC
 
     public override async Task<bool> Execute(TaskExecutionLog currentJob)
     {
-        var bounces = await GetLatestSuppressionsByType<BlockOrBounceDto>("bounces");
-
-        foreach (var bounce in bounces)
-        {
-            await contactService.Unsubscribe(bounce.Email, $"{bounce.Status} - {bounce.Reason}", "SendGrid_bounces", bounce.CreatedAt, null);
-        }
-
-        var blocks = await GetLatestSuppressionsByType<BlockOrBounceDto>("blocks");
-
-        foreach (var block in blocks)
-        {
-            await contactService.Unsubscribe(block.Email, $"{block.Status} - {block.Reason}", "SendGrid_blocks", block.CreatedAt, null);
-        }
-
-        var spamReports = await GetLatestSuppressionsByType<SpamReportDto>("spam_reports");
+        await Unsubscribe<BlockOrBounceDto>("bounces");
+        await Unsubscribe<BlockOrBounceDto>("blocks");
+        await Unsubscribe<SpamReportDto>("spam_reports");
+        await Unsubscribe<SuppressionDto>("unsubscribes");       
+        return true;
+    }
 
-        foreach (var spamReport in spamReports)
-        {
-            await contactService.Unsubscribe(spamReport.Email, "Reported As Spam", "SendGrid_spam_reports", spamReport.CreatedAt, spamReport.Ip);
-        }
+    private string GetSecondaryApiKeyName(int i)

```

Comment:
Do you think this method is really needed here? Since it's a simple string concatenation, you can remove the method and replace its usage with inline string interpolation.